ttnn/cpp/ttnn/operations/moreh/moreh_nll_loss_backward/device/moreh_nll_loss_backward_device_operation.cpp

```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_nll_loss_backward_device_operation.hpp"
+
+namespace ttnn::operations::moreh::moreh_nll_loss_backward {
+
+MorehNllLossBackwardDeviceOperation::program_factory_t MorehNllLossBackwardDeviceOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    return Factory{};
+}
+
+void MorehNllLossBackwardDeviceOperation::validate_inputs(
+    const operation_attributes_t& attributes, const tensor_args_t& tensor_args) {
+    auto& target_tensor = tensor_args.target_tensor;
+    auto& output_grad_tensor = tensor_args.output_grad_tensor;
+    auto& weight_tensor = tensor_args.weight_tensor;
+    auto& divisor_tensor = tensor_args.divisor_tensor;
+    auto& input_grad_tensor = tensor_args.input_grad_tensor;
+
+    TT_FATAL(target_tensor.storage_type() == StorageType::DEVICE, "Operands to nll_loss need to be on device!");
+    TT_FATAL(target_tensor.buffer() != nullptr, "Operands to nll_loss need to be allocated in buffers on device!");
+    TT_FATAL((target_tensor.get_layout() == Layout::TILE), "target_tensor to nll_loss must be tilized");
+    TT_FATAL(target_tensor.get_dtype() == DataType::INT32);
```

Comment:
please add an error description here, include current value.
description strings allow to inject values with {}, value is provided after comma
```cpp
TT_FATAL(condition, description, values_to_inject)
```

Reply:
I have added the description.